corehq/apps/userreports/app_manager/helpers.py

```diff
@@ -74,14 +75,30 @@ def get_form_data_source(app, form):
     ]
     dynamic_properties = _get_dynamic_indicators_from_export_schema(schema)
     form_name = form.default_name()
-    return DataSourceConfiguration(
+    config = DataSourceConfiguration(
         domain=app.domain,
         referenced_doc_type='XFormInstance',
         table_id=clean_table_name(app.domain, form_name),
         display_name=form_name,
         configured_filter=make_form_data_source_filter(xform.data_node.tag_xmlns, app.get_id),
         configured_indicators=meta_properties + dynamic_properties + _get_shared_indicators(),
     )
+    return _deduplicate_columns_if_necessary(config)
```

Comment:
If there are already data sources with conflicting column ids, then this would trigger a rebuild on each of those right?

Before deploying this it would be good to know:
- How many data sources with duplicate columns there are? 
- How large are they?
- How long will it take to rebuild each of them?
- Will it significantly impact their reports?

Reply:
As best as I can tell this code is only used when initially creating a data source via a page that hardly anyone uses so I don't think it would affect any existing data sources or trigger any rebuilds.

Also the failure mode of this was that the page 500s when trying to create the data source so I don't think there should be any data sources in this duplicate column state (at least as a result of this bug) as they would fail validation.

Reply:
Thanks. That makes sense.

Would be really nice if we could separate out the backend data source/reporting code from the HQ integration and UI layers to make this separation clearer, but I imagine that will end up being a lot of work

Reply:
agreed. i think in my ideal world UCR data sources would be a library but seems a hard path to get there.